.github/unittest.sh

```diff
@@ -7,6 +7,30 @@ CONDA_PATH=$(which conda)
 eval "$(${CONDA_PATH} shell.bash hook)"
 conda config --set channel_priority strict
 
+# Setup the OS_TYPE environment variable that should be used for conditions involving the OS below.
+case $(uname) in
+  Linux)
+    OS_TYPE=linux
+    ;;
+  Darwin)
+    OS_TYPE=macos
+    ;;
+  *)
+    echo "Unknown OS type:" $(uname)
+    exit 1
+    ;;
+esac
+
+# The x86 macOS runners, e.g. the GitHub Actions native "macos-12" runner, has some JPEG libraries installed by default
+# that interfere with our build. We uninstall them here and use the one from conda below.
+if [[ "${OS_TYPE}" == "macos" && $(uname -m) == x86_64 ]]; then
+  JPEG_LIBS=$(brew list | grep jpeg)
+  echo $JPEG_LIBS
+  for lib in $JPEG_LIBS; do
+    brew uninstall --ignore-dependencies --force $lib || true
+  done
+fi
```

Comment:
We are doing the same for the macOS wheel workflows:

https://github.com/pytorch/vision/blob/0e62c34fb6ed7766fea368e67033564d8010ea78/packaging/pre_build_script.sh#L3-L9